tests/test_nzbstuff.py

```diff
@@ -50,6 +50,52 @@ def test_nzo_basic(self):
         # TODO: More checks!
 
 
+class Server:
+    def __init__(self, host, priority, active):
+        self.host = host
+        self.priority = priority
+        self.active = active
+
+
+class TestArticle:
+    def test_get_article(self):
+        article_id = "test@host" + os.urandom(8).hex() + ".sab"
+        article = nzbstuff.Article(article_id, randint(4321, 54321), None)
+        servers = []
+        servers.append(Server("testserver1", 10, True))
+        servers.append(Server("testserver2", 20, True))
+        servers.append(Server("testserver3", 30, True))
+
+        # Test fetching top priority server
+        server = servers[0]
+        assert article.get_article(server, servers) == article
+        assert article.fetcher_priority == 10
+        assert article.fetcher == server
+        assert article.get_article(server, servers) == None
+        article.fetcher = None
+        article.add_to_try_list(server)
+        assert article.get_article(server, servers) == None
+
+        # Test fetching when there is a higher priority server available
+        server = servers[2]
+        assert article.fetcher_priority == 10
+        assert article.get_article(server, servers) == None
+        assert article.fetcher_priority == 20
+
+        # Server should be used even if article.fetcher_priority is a higher number than server.priority
+        article.fetcher_priority = 30
+        server = servers[1]
+        assert article.get_article(server, servers) == article
```

Comment:
I ran the unittest on the old code and then it fails at this line. So, what is changed? I don't fully understand to be honest.

Reply:
Ah I understand, but now I wonder why the old code didn't do this... Or is this a situation that shouldn't happen.

Reply:
I suspected that there might be a way to have the priority of all the remaining articles set such that after changing the priority of an untried server to a lower priority number than the articles, it would end up being skipped because the old version would skip it. 

I haven't been able to create it in a real world test so maybe there is something I've missed that avoids such situations. It's still a bug if it can happen because then the lower pri server would probably continue downloading the articles even though there would be a higher pri server available.